File: Frontend/test_route.js
const fetch = require('node-fetch'); // wait node 18+ has fetch natively

async function test() {
  const origin = [77.5946, 12.9716];
  const destination = [76.6499, 12.2958];
  const from = `${origin[0]},${origin[1]}`;
  const to = `${destination[0]},${destination[1]}`;
  const params = new URLSearchParams({
    alternatives: 'true',
    steps: 'true',
    geometries: 'geojson',
    overview: 'full'
  });
  
  const url = `https://router.project-osrm.org/route/v1/driving/${from};${to}?${params}`;
  console.log("URL:", url);
  try {
      const response = await fetch(url);
      const data = await response.json();
      if (!data.routes) {
        console.log("NO ROUTES", data);
        return;
      }
      console.log(`Got ${data.routes.length} routes.`);
  } catch (e) {
      console.log("Error:", e);
  }
}
test();

File: Frontend/test_route.py
import urllib.request
import json
url = "https://router.project-osrm.org/route/v1/driving/77.5946,12.9716;76.6499,12.2958?alternatives=true&steps=true&geometries=geojson&overview=full"
try:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        if 'routes' in data:
            print(f"Got {len(data['routes'])} routes.")
        else:
            print("No routes found:", data)
except Exception as e:
    print(f"Error: {e}")
